src/aiida/orm/nodes/data/remote/stash/base.py

```diff
@@ -39,13 +39,16 @@ class RemoteStashData(Data):
     class Model(Data.Model):
         stash_mode: StashMode = MetadataField(description='The mode with which the data was stashed')
 
-    def __init__(self, stash_mode: StashMode, **kwargs):
+    def __init__(self, stash_mode: StashMode, source_uuid: str, **kwargs):
         """Construct a new instance
 
         :param stash_mode: the stashing mode with which the data was stashed on the remote.
+        :param source_uuid: the uuid of the source node (str). If the source node itself is a stashed node,
+            then the `source_uuid` of the stashed node should be passed.
         """
         super().__init__(**kwargs)
         self.stash_mode = stash_mode
+        self.source_uuid = source_uuid
```

Comment:
Shouldn't we add a check that this node exists and is a RemoteData ore RemoteStashData?